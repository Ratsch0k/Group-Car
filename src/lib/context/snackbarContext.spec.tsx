import React, { useEffect } from 'react';
import { render, waitFor, screen } from "@testing-library/react";
import SnackbarProvider, { SnackbarContext, SnackbarType } from "./snackbarContext";

describe('show snackbar with type', () => {
  ['success', 'warning', 'error', 'info'].forEach((type) => {
    [true, false].forEach((withClose) => {
      describe(`${type} and withClose set to `, () => {
        it(`${withClose} is rendered correctly`, async () => {
          let renderCount = 0;

          const {baseElement} = render(
            <SnackbarProvider>
              <SnackbarContext.Consumer>
                {({show}) => {
                  if (renderCount == 0) {
                    show({
                      type: type as SnackbarType,
                      withClose,
                      content: 'SNACKBAR',
                    });
                  }
                  renderCount++;
                  

                  return (
                    <div>{type}</div>
                  );
                }}
              </SnackbarContext.Consumer>
            </SnackbarProvider>
          );

          await waitFor(() => expect(screen.queryByText('SNACKBAR')).toBeDefined());
          
          expect(baseElement).toMatchSnapshot();
        });
      });
    });
  });
});
