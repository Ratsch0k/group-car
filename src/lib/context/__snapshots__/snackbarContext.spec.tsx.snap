// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click on close closes the snackbar 1`] = `
<body>
  <div>
    <div>
      TEST
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`click on close closes the snackbar 2`] = `
<body>
  <div>
    <div>
      TEST
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiSnackbarContent-root MuiPaper-elevation6"
        direction="up"
        role="alert"
        style="opacity: 0; transform: scale(0.75, 0.5625); visibility: hidden;"
      >
        <div
          class="MuiSnackbarContent-message"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`renders action correctly 1`] = `
<body>
  <div>
    <div>
      TEST
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              BUTTON
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type error and withClose set to  false is rendered correctly 1`] = `
<body>
  <div>
    <div>
      error
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardError makeStyles-error-18 MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type error and withClose set to  true is rendered correctly 1`] = `
<body>
  <div>
    <div>
      error
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardError makeStyles-error-15 MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type info and withClose set to  false is rendered correctly 1`] = `
<body>
  <div>
    <div>
      info
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type info and withClose set to  true is rendered correctly 1`] = `
<body>
  <div>
    <div>
      info
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardInfo MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11,9H13V7H11M12,20C7.59,20 4,16.41 4,12C4,7.59 7.59,4 12,4C16.41,4 20,7.59 20, 12C20,16.41 16.41,20 12,20M12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0,0 22,12A10, 10 0 0,0 12,2M11,17H13V11H11V17Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type success and withClose set to  false is rendered correctly 1`] = `
<body>
  <div>
    <div>
      success
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess makeStyles-success-4 MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type success and withClose set to  true is rendered correctly 1`] = `
<body>
  <div>
    <div>
      success
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardSuccess makeStyles-success-1 MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M20,12A8,8 0 0,1 12,20A8,8 0 0,1 4,12A8,8 0 0,1 12,4C12.76,4 13.5,4.11 14.2, 4.31L15.77,2.74C14.61,2.26 13.34,2 12,2A10,10 0 0,0 2,12A10,10 0 0,0 12,22A10,10 0 0, 0 22,12M7.91,10.08L6.5,11.5L11,16L21,6L19.59,4.58L11,13.17L7.91,10.08Z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type warning and withClose set to  false is rendered correctly 1`] = `
<body>
  <div>
    <div>
      warning
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardWarning makeStyles-warning-11 MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`show snackbar with type warning and withClose set to  true is rendered correctly 1`] = `
<body>
  <div>
    <div>
      warning
    </div>
    <div
      class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter"
    >
      <div
        class="MuiPaper-root MuiAlert-root MuiAlert-standardWarning makeStyles-warning-8 MuiPaper-elevation0"
        direction="up"
        role="alert"
        style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message"
        >
          SNACKBAR
        </div>
        <div
          class="MuiAlert-action"
        >
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeSmall"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiIconButton-label"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
